src/main/java/io/aiven/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -134,27 +108,77 @@ public BufferedRecords(
         return flushed;
     }
 
+    private void prepareStatement() throws SQLException {
+        final String sql = writeSql();
+
+        log.debug("Prepared SQL {} for insert mode {}", sql, config.insertMode);
+
+        close();
+        preparedStatement = connection.prepareStatement(sql);
+        preparedStatementBinder = dbDialect.statementBinder(
+                preparedStatement,
+                config.pkMode,
+                currentSchemaPair,
+                fieldsMetadata,
+                config.insertMode
+        );
+    }
+
+    private String writeSql() {
+        final String sql;
+        log.debug("Generating query for insert mode {} and {} records", config.insertMode, records.size());
+        if (config.insertMode == MULTI) {
+            sql = getMultiInsertSql(tableDefinition);
```

Comment:
`tableDefinition` is now a field in the object, no need to pass it as a method argument.

Reply:
And I didn't notice that it's actually unused in the method...